packages/material-ui-unstyled/src/FormControlUnstyled/FormControlUnstyled.tsx

```diff
@@ -0,0 +1,257 @@
+          ].join('\n'),
+        );
+      }
+
+      registeredInput.current = true;
+      return () => {
+        registeredInput.current = false;
+      };
+    };
+  }
+
+  const onFilled = React.useCallback(() => {
+    setFilled(true);
+  }, []);
+
+  const onEmpty = React.useCallback(() => {
+    setFilled(false);
+  }, []);
+
+  const childContext: FormControlState = {
+    adornedStart,
+    setAdornedStart,
+    disabled,
+    error,
+    filled,
+    focused,
+    hiddenLabel,
+    onBlur: () => {
+      setFocused(false);
+    },
+    onEmpty,
+    onFilled,
+    onFocus: () => {
+      setFocused(true);
+    },
+    registerEffect,
+    required,
+    ...extraContextProperties,
+  };
+
+  const Root = component ?? components.Root ?? 'div';
+  const rootProps = appendStyleProps(Root, { ...other, ...componentsProps.root }, styleProps);
+
+  return (
+    <FormControlContext.Provider value={childContext}>
+      <Root
+        ref={ref}
+        {...rootProps}
+        className={clsx(
+          classes.root,
+          className,
+          rootProps?.className,
+          disabled && classes.disabled,
+        )}
+      >
+        {children}
+      </Root>
+    </FormControlContext.Provider>
+  );
+});
+
+FormControl.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * The content of the component.
+   */
+  children: PropTypes.node,
+  /**
+   * Class name applied to the root element.
+   */
+  className: PropTypes.string,
+  /**
+   * The component used for the Root slot.
+   * Either a string to use a HTML element or a component.
+   * This is equivalent to `components.Root`. If both are provided, the `component` is used.
+   */
+  component: PropTypes.elementType,
+  /**
+   * The components used for each slot inside the FormControl.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  components: PropTypes.shape({
+    Root: PropTypes.elementType,
+  }),
+  /**
+   * @ignore
+   */
+  componentsProps: PropTypes.object,
+  /**
+   * If `true`, the label, input and helper text should be displayed in a disabled state.
+   * @default false
+   */
+  disabled: PropTypes.bool,
+  /**
+   * If `true`, the label is displayed in an error state.
+   * @default false
+   */
+  error: PropTypes.bool,
+  /**
+   * Extra properties to be placed on the FormControlContext.
+   * @default {}
+   */
+  extraContextProperties: PropTypes.object,
+  /**
+   * If `true`, the component is displayed in focused state.
+   * @default false
+   */
+  focused: PropTypes.bool,
+  /**
+   * If `true`, the label is hidden.
+   * This is used to increase density for a `FilledInput`.
```

Comment:
This is a Material-UI component. It seems like this prop is unrelated to an "unstyled" version of the FormControl.